### PR TITLE
Integrate type tests gen and fix non-conformant scripts

### PR DESCRIPTION
**Pull request by anthony-murphy**
This change integrates type test generation into the build. With fluid-build and build:fast type test generation will only run when the package json changes, as part of #9914. This works as we only need to regenerate when the version changes, or when the broken section changes. This adds minimal overhead to most incremental builds.

### CODE REVIEW

**Comment by anthony-murphy** [comment]
working through expected automation issue in draft form

File: packages/framework/fluid-framework/src/test/tsconfig.json [hunk 1]
```diff
@@ -1,19 +0,0 @@
-{
-    "extends": "@fluidframework/build-common/ts-common-config.json",
-    "compilerOptions": {
-        "rootDir": "./",
-        "outDir": "../../dist/test",
-        "declaration": false,
-        "declarationMap": false,
-        "skipLibCheck": true,
```
Comment (heliocliu) [inline packages/framework/fluid-framework/src/test/tsconfig.json:8]: was the skipLibCheck stuff fixed?

Comment (heliocliu) [reply packages/framework/fluid-framework/src/test/tsconfig.json:8]: oh nvm the typetest stuff is disabled here ok

Comment (anthony-murphy) [reply packages/framework/fluid-framework/src/test/tsconfig.json:8]: this package doesn't have type tests right now, as there are problems with const enums coming from the merge tree package. this is something that need to be fix #8142. So this was tsconfig for nothing that was just causing problems

**Review by heliocliu — APPROVED** [review]

**Pull request merged**